migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    create_user                         character varying(60)                   NOT NULL,
+    create_timestamp                    timestamp with time zone DEFAULT now()  NOT NULL,
+    update_user                         character varying(60)                   NOT NULL,
+    update_timestamp                    timestamp with time zone DEFAULT now()  NOT NULL
+);
+
+COMMENT ON TABLE mine_report_submission_status_code is 'All the possible available status codes for mine reports.';
+ALTER TABLE mine_report_submission_status_code OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_definition (
+    mine_report_definition_id            serial                                  PRIMARY KEY, 
+    mine_report_definition_guid          uuid                                    NOT NULL   ,
+    name                                 varchar(50)                             NOT NULL   ,
+    description                          varchar(300)                            NOT NULL   ,
+    compliance_article_id                integer                                 NOT NULL   ,
+    due_date_period_months               integer                                 NOT NULL   ,
+    mine_report_due_date_type            character varying(3)                    NOT NULL   ,
+    active_ind                           boolean DEFAULT true                    NOT NULL   ,
+    create_user                          character varying(60)                   NOT NULL   ,
+    create_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+    update_user                          character varying(60)                   NOT NULL   ,
+    update_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+
+    FOREIGN KEY (compliance_article_id) REFERENCES compliance_article(compliance_article_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_due_date_type) REFERENCES mine_report_due_date_type(mine_report_due_date_type) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_definition is 'Definitions of code required reports, due dates and articles of the code that apply to them.';
+ALTER TABLE mine_report_definition OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report
+(
+    mine_report_id               serial                                   NOT NULL PRIMARY KEY,
+    mine_report_guid             uuid                                     NOT NULL            ,
+    mine_report_definition_id    integer                                  NOT NULL            ,
+    mine_guid                    uuid                                     NOT NULL            ,
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    submission_year              integer                                                      ,
```

Comment:
Will this column be sized to 4 characters? Or should it be? Is this a free form field? Will there be any specific reporting off this column?   ....wondering if this column should be driven by a lookup table, or find a way to ensure the values are consistently entered.

Reply:
This is used on one other table which is where I got this and it has no limits at the moment. I think we will need some good backend validation for it.